apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetItemResultMapper.java

```diff
@@ -0,0 +1,237 @@
+
+            Map<String, JsonNode> data = new HashMap<>(getData(row));
+
+            JsonNode input = getJsonNode(row, data, "input");
+            JsonNode expectedOutput = getJsonNode(row, data, "expected_output");
+            JsonNode metadata = getJsonNode(row, data, "metadata");
+
+            if (!data.containsKey("input")) {
+                data.put("input", input);
+            }
+
+            if (!data.containsKey("expected_output")) {
+                data.put("expected_output", expectedOutput);
+            }
+
+            if (!data.containsKey("metadata")) {
+                data.put("metadata", metadata);
+            }
+
+            return DatasetItem.builder()
+                    .id(row.get("id", UUID.class))
+                    .input(input)
+                    .data(new HashMap<>(data))
+                    .expectedOutput(expectedOutput)
+                    .metadata(metadata)
+                    .source(DatasetItemSource.fromString(row.get("source", String.class)))
+                    .traceId(Optional.ofNullable(row.get("trace_id", String.class))
+                            .filter(s -> !s.isBlank())
+                            .map(UUID::fromString)
+                            .orElse(null))
+                    .spanId(Optional.ofNullable(row.get("span_id", String.class))
+                            .filter(s -> !s.isBlank())
+                            .map(UUID::fromString)
+                            .orElse(null))
+                    .experimentItems(getExperimentItems(row.get("experiment_items_array", List[].class)))
+                    .lastUpdatedAt(row.get("last_updated_at", Instant.class))
+                    .createdAt(row.get("created_at", Instant.class))
+                    .createdBy(row.get("created_by", String.class))
+                    .lastUpdatedBy(row.get("last_updated_by", String.class))
+                    .build();
+        });
+    }
+
+    static Map<String, JsonNode> getData(Row row) {
+        return Optional.ofNullable(row.get("data", Map.class))
+                .filter(s -> !s.isEmpty())
+                .map(value -> (Map<String, String>) value)
+                .stream()
+                .map(Map::entrySet)
+                .flatMap(Collection::stream)
+                .map(entry -> Map.entry(entry.getKey(), JsonUtils.getJsonNodeFromString(entry.getValue())))
+                .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    static JsonNode getJsonNode(Row row, Map<String, JsonNode> data, String key) {
+        JsonNode json = null;
+
+        if (data.containsKey(key)) {
+            json = data.get(key);
+        }
+
+        if (json == null) {
+            json = Optional.ofNullable(row.get(key, String.class))
+                    .filter(s -> !s.isBlank())
+                    .map(JsonUtils::getJsonNodeFromString).orElse(null);
+        }
+
+        return json;
+    }
+
+    static String getJsonNodeOrDefault(JsonNode jsonNode) {
```

Comment:
Minor: all these methods are overloaded versions, which receive a different type. I'd go with the previous naming of `getOrDefault` instead of adding the type to the method name.